### PR TITLE
feat(workflow): add github-repo-stats file

### LINKED ISSUE #12752: [GH Repo Stats]: Configure tool
### User story

> As a `product manager for Carbon`:

> I need to: track what our consumers are using as it relates to the monorepo

> so that I can: make more informed decisions on the direction of Carbon.

### Additional information

- _https://github.com/marketplace/actions/github-repo-stats_
- _https://github.com/jgehrcke/github-repo-stats/wiki/Tutorial_

### Acceptance criteria

- [ ] _GH Repo stats is properly configured on the monorepo to capture stats beyond the 15 day limit._

### PR DESCRIPTION
**Pull request by sstrubberg**
Closes #12752 

Addes a workflow file to collect repo stats beyond the 15 day limit

https://github.com/jgehrcke/github-repo-stats/wiki/Tutorial#step-2-add-workflow-file-to-default-branch

### CODE REVIEW

**Review by francinelucca — APPROVED** [review]
LGTM!
should we also do step 3 here so that the `github-repo-stats` branch this is going to push to just contains the stats files? https://github.com/jgehrcke/github-repo-stats/wiki/Tutorial#step-3-prepare-data-branch-trigger-first-run

File: .github/workflows/github-repo-stats.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: github-repo-stats
+
+on:
+  schedule:
+    # Run this once per day, towards the end of the day for keeping the most
+    # recent data point most meaningful (hours are interpreted in UTC).
+    - cron: '0 23 * * *'
```
Comment (tay1orjones) [inline .github/workflows/github-repo-stats.yml]: This is 3pm PDT - what if we did 01:00 or 02:00 UTC instead to capture the end of the day all the way over in pacific time?

**Review by tay1orjones — APPROVED** [review]
Looks good, only one suggestion. I agree with Francine on clearing out the branch 👍

**Comment by sstrubberg** [comment]
> LGTM! should we also do step 3 here so that the `github-repo-stats` branch this is going to push to just contains the stats files? https://github.com/jgehrcke/github-repo-stats/wiki/Tutorial#step-3-prepare-data-branch-trigger-first-run

yes! i'm having a hard time clearing out the branch though. Every time I try to make a PR against the monorepo it says there are no changes.

**Comment by jgehrcke** [comment]
Hello @sstrubberg  @tay1orjones @francinelucca -- sorry for sneaking up there. I am the author of github-repo-stats and always curious about new use cases. I see that the GHA is failing in your case with this error message:
```
github.GithubException.GithubException: 403 {"message": "Resource protected by organization SAML enforcement. You must grant your Personal Access token access to this organization.", "documentation_url": "https://docs.github.com/articles/authenticating-to-a-github-organization-with-saml-single-sign-on/"}
```
Indeed, [GitHub docs](https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-saml-single-sign-on/authorizing-a-personal-access-token-for-use-with-saml-single-sign-on) seem to explain what needs to be done. Hope you will get this running. Let me know if I can help.

**Comment by tay1orjones** [comment]
Thanks for the heads up @jgehrcke 

@sstrubberg I updated the token, the workflow's succeeding now 👍

**Pull request merged**